orangecanvas/document/quickmenu.py

```diff
@@ -752,6 +900,30 @@ def __updateTab(self, index):
         if tab.palette:
             b.setPalette(tab.palette)
 
+    def __updateShadows(self):
+        currentIndex = self.currentIndex()
+
+        buttons = [tab.button for tab in self.__tabs if tab.button.isVisibleTo(self.parent())]
+        if not buttons:
+            return
+
+        # set left/right shadows
+        for b in buttons:
+            b.shadowPosition = 10
+
+        # set top shadow
+        buttons[0].shadowPosition |= 1
+        # set bottom shadow
+        buttons[-1].shadowPosition |= 4
+
+        # set chosen tab shadows
+        if currentIndex + 1 < len(self.__tabs):
+            self.__tabs[currentIndex + 1].button.shadowPosition |= 1  # top shadow
+        if currentIndex - 1 >= 0:
+            self.__tabs[currentIndex - 1].button.shadowPosition |= 4  # bottom shadow
+
+        self.update()
```

Comment:
`update` should be called on all buttons that have had their display properties changed.